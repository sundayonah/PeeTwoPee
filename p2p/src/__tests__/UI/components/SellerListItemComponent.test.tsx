import { MockedProvider } from '@apollo/client/testing';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SellerListItemComponent from "../../../pages/Sell/components/SellListData/SellListItemComponent"


describe('<SellerListItemComponent />',()=>{
    const checkForConnection = () => {
    
    }
   const priceObject = {
    _id: "993",
    type: "sell",
    asset: "BUSD",
    fiat: "NGN",
    price_type: "string",
    price: "23",
    crypto_amount: 3,
    sold: 1,
    payment_method: "bank tranfer",
    limit_min: 7,
    limit_max: 10,
    duration: 12,
    terms: "string",
    status: "string",
    auto_reply: "",
    createdAt: "string",
    updatedAt: "string",
    user: {
        fullname:"string",
    }
      }
  it("has the ads information",async ()=>{
    const { container } = render(
     <SellerListItemComponent data={priceObject} page="buy" checkForConnection={checkForConnection} />   
    )
    const price = screen.getByText(priceObject.Price);
    const paymentMethod = screen.getByText(priceObject['Payment Method']);
    const available = screen.getByTestId("available")
    const limit = screen.getByTestId("limit")
    const button = screen.getByTestId("button")
    

    expect(price).toBeInTheDocument()
    expect(paymentMethod).toBeInTheDocument()
    expect(available).toHaveTextContent(`${priceObject.amount} ${priceObject.token}`)
    expect(limit).toHaveTextContent(`NGN${priceObject.Limits[0]} - NGN${priceObject.Limits[1]}`)
    expect(button).toHaveStyle(`background:#0CCB80`)
    expect(button).toHaveTextContent(`Buy ${priceObject.token}`)

     userEvent.click(button)
  })

  it("Matches snapshot",()=>{
    const {container} = render(<SellerListItemComponent priceObject={priceObject} page="Buy" checkForConnection={checkForConnection} />)
    expect(container).toMatchSnapshot()
})
})