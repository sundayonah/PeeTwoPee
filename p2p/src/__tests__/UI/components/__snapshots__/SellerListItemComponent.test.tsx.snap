// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerListItemComponent /> Matches snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-11uwsgz"
    >
      <div
        class="css-0"
      >
        <div
          class="css-k008qs"
        >
          <div
            class="css-1f52nmj"
            data-testid="profilePicture"
          >
            G
          </div>
          <div
            class="css-9n3bbc"
          >
            <div
              class="css-1euf48e"
            >
              <p
                class="chakra-text css-5r24gd"
                data-testid="profileUsername"
              >
                GabrielS
              </p>
              <div
                class="css-m0ceft"
                data-testid="profileStatus"
              />
              <img
                class="chakra-image css-14gt3bc"
                data-testid="profileBadge"
                src=""
              />
            </div>
            <p
              class="chakra-text css-1vzthj9"
              data-testid="profileInformation"
            >
              45
               Order(s) | 
              34.6
              % Completion
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-k008qs"
        >
          <p
            class="chakra-text css-gre1pd"
          >
            500.00
          </p>
          <p
            class="chakra-text css-1ol4k66"
          >
            NGN
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-gg4vpm"
        >
          <p
            class="chakra-text css-141h1cx"
          >
            Available
          </p>
          <p
            class="chakra-text css-hhrkbn"
            data-testid="available"
          >
            0.045
             
            BNB
          </p>
        </div>
        <div
          class="css-gg4vpm"
        >
          <p
            class="chakra-text css-ae7sfc"
          >
            Limit
          </p>
          <p
            class="chakra-text css-hhrkbn"
            data-testid="limit"
          >
             NGN
            3000.00
             - NGN
            53000.00
          </p>
        </div>
      </div>
      <div
        class="css-j93o2j"
      >
        <div
          class="css-1l4w6pd"
        >
          <div
            class="css-v8h3im"
          >
            Bank Transfer
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-1bvc4cc"
        >
          <button
            class="chakra-button css-18nrrwh"
            data-testid="button"
            type="button"
          >
            Sell BNB
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
